### ISSUE TITLE
no-top-level-browser-globals rule is flagging types as top level browser globals

### ISSUE THREAD

**Issue by machadinhos** [sveltejs/eslint-plugin-svelte#1226]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.27.0

### What version of `eslint-plugin-svelte` are you using?

3.8.0

### What did you do?

```svelte
<script lang="ts">
  let element: HTMLElement;
  let submitHandler: SubmitEvent = (): void => {};
</script>

<form onsubmit={submitHandler}></form>
<div bind:this={element}></div>
```

### What did you expect to happen?

It shouldn't flag types as top level browser globals, because, from my understanding, the objective of this rule is to prevent calling browser specific APIs (e.g. `localStorage`) while in the server.

### What actually happened?

The new `no-top-level-browser-globals` rule is flagging types as top level browser globals.

### Link to **GitHub Repo** with Minimal Reproducible Example

N.A.

**Comment by that-richan** [comment]
This bug seems to have been reintroduced.

**Comment by baseballyama** [comment]
Please make a new issue with proper repl please.

**Comment by machadinhos** [comment]
> This bug seems to have been reintroduced.

Hi!

I couldn't replicate with the same example I gave when opening this issue.

@that-richan  are you sure you are using the latest version of the plugin?
eslint-plugin-svelte@3.12.3

**Comment by that-richan** [comment]
> > This bug seems to have been reintroduced.
> 
> Hi!
> 
> I couldn't replicate with the same example I gave when opening this issue.
> 
> [@that-richan](https://github.com/that-richan) are you sure you are using the latest version of the plugin? eslint-plugin-svelte@3.12.3

<img width="1536" height="104" alt="Image" src="https://github.com/user-attachments/assets/5eb03f67-9272-4704-b838-d45807a1ca61" />

Ok seems to be a problem only inside generics?

```
let element = $state<HTMLDivElement | null>(null); // has a problem
let element: HTMLDivElement | null = null; // no problem
```
will file a new issue.

**Comment by that-richan** [comment]
https://github.com/sveltejs/eslint-plugin-svelte/issues/1361
